Sources/Bow/Data/Day.swift

```diff
@@ -4,29 +4,93 @@ public final class ForDay {}
 public final class DayPartial<F: Comonad, G: Comonad>: Kind2<ForDay, F, G> {}
 public typealias DayOf<F: Comonad, G: Comonad, A> = Kind<DayPartial<F, G>, A>
 
-public class Day<F: Comonad, G: Comonad, A> : DayOf<F, G, A> {
-    public static func fix(_ value : DayOf<F, G, A>) -> Day<F, G, A> {
-        return value as! Day<F, G, A>
+public final class Day<F: Comonad, G: Comonad, A> : DayOf<F, G, A> {
+    internal let left: Kind<F, Any /*B*/>
+    internal let right: Kind<G, Any /*C*/>
+    internal let f: (Any /*B*/, Any /*C*/) -> A
+
+    public init(left: Kind<F, Any>,
```

Comment:
we could restrict unless the constructor to match the types, although we eare them internally.

example:
```swift
public init<B, C>(left: Kind<F, B>,
                right: Kind<G, C>,
                _ f: @escaping (B, C) -> A) {
        self.left = left.map { b in b as Any }
        self.right = right.map { c in c as Any }
        self.f = { (b: Any, c: Any) in f(b as! B, c as! C) }
    }
```